### ISSUE TITLE
Add configuration file support besides existing command-line/env options

### ISSUE THREAD

**Issue by c2hpxq** [AFLplusplus/AFLplusplus#1250]
Tools in AFL++ make heavy use of argv&envp. I think as the project evolves, it gradually becomes a bad idea. Settings scatter everywhere and not in a uniform way
Somewhat similar to Syzkaller, how about write all the configurations in a single config file (maybe in the JSON format?) and pass the file to tools? 
I think that might also facilitate communication, since people can then share their settings by simply publishing a single file.

**Comment by vanhauser-thc** [comment]
Personally I want to start a fuzz instance and not open an editor first. This would also makes scripting and automation much harder.
normally you have several instances running in sync, removing one after some time and replacing with something else. with a config file this is extra work.

I do see the complication with many command line options and many more environment variables, where a config could potentially make this easier. 

So I myself would not put effort into this.
But I would accept a PR making this optionally available (e.g. `afl-fuzz -J config.json`).

dunno what the other in the team think about this.

**Comment by c2hpxq** [comment]
Yes, I agree with you. Backward compatibility does matter and I think it's important to keep the 'out-of-the-box' merit of AFL.
For a tool as popular, good as AFL++ ,this must be optional at first and any further decision should be based on the feedback of fuzzing community.

**Comment by domenukk** [comment]
I agree that this could be an add-on /alternative to the current way to set up a run.

However, I'm not sure this needs to be part of the core tool.
Basically, you can already have a wrapper script (python?) that sets all of the appropriate configs variables and then calls the `afl-fuzz` binary with the given parameters. 
But maybe I'm just too used to the way things are...

(Speaking of the way things are, are you aware that afl-fuzz stores the env variables and command line in the output folder, and you can also share these?)

From an implementation point of view, incorporating this into the core `afl-fuzz` means we would need some sort of json/toml/ini parser in the project, which we do not have at the moment (unless we call out to python, if it's linked)
Then there are all of the surrounding tools like afl-cmin, tmin, showmap, that would also need to, either, understand this config file, or take their own configs.

TL;DR this is going to be some work, but might be fun to have if it's done right

**Comment by domenukk** [comment]
Feel free to provide a PR if you're still interested in this issue. Closing for now
